docs/source/argument_check.rst

```diff
@@ -0,0 +1,607 @@
+
+.. code-block:: bash
+
+      DOUBLE PRECISION FUNCTION DASUM(N,DX,INCX)
+      IF (N.LE.0 .OR. INCX.LE.0) RETURN
+
+      DOUBLE PRECISION FUNCTION DNRM2(N,X,INCX)
+      IF (N.LT.1 .OR. INCX.LT.1) THEN
+          return = ZERO
+
+      SUBROUTINE DSCAL(N,DA,DX,INCX)
+      IF (N.LE.0 .OR. INCX.LE.0) RETURN
+
+      INTEGER FUNCTION IDAMAX(N,DX,INCX)
+      IDAMAX = 0
+      IF (N.LT.1 .OR. INCX.LE.0) RETURN
+      IDAMAX = 1
+      IF (N.EQ.1) RETURN
+
+Below is current rocblas_scal_strided_batched_impl
+
+.. code-block:: c++
+
+    template <rocblas_int NB, typename T, typename U>
+    rocblas_status rocblas_scal_strided_batched_impl(rocblas_handle handle,
+                                                     rocblas_int    n,
+                                                     const U*       alpha,
+                                                     T*             x,
+                                                     rocblas_int    incx,
+                                                     rocblas_stride stridex,
+                                                     rocblas_int    batch_count)
+    {
+        if(!handle)
+            return rocblas_status_invalid_handle;
+        if(!alpha)
+            return rocblas_status_invalid_pointer;
+
+        auto layer_mode = handle->layer_mode;
+        ------snip---logging---code---does-not-dereference---NULL---pointer---------------------------------------------------
+        if(!x)
+            return rocblas_status_invalid_pointer;
+
+        if(batch_count < 0)
+            return rocblas_status_invalid_size;
+
+        if(n <= 0 || incx <= 0 || batch_count <= 0)
+        {
+            return rocblas_status_success;
+        }
+
+We need to change this to
+
+.. code-block:: c++
+
+    template <rocblas_int NB, typename T, typename U>
+    rocblas_status rocblas_scal_strided_batched_impl(rocblas_handle handle,
+                                                     rocblas_int    n,
+                                                     const U*       alpha,
+                                                     T*             x,
+                                                     rocblas_int    incx,
+                                                     rocblas_stride stridex,
+                                                     rocblas_int    batch_count)
+    {
+        if(!handle)
+            return rocblas_status_invalid_handle;
+  //      if(!alpha)
+  //          return rocblas_status_invalid_pointer;
+
+        auto layer_mode = handle->layer_mode;
+ ------snip---logging---code---does-not-dereference---NULL---pointer--------------------------------------------------
+
+  //      if(!x)
+  //          return rocblas_status_invalid_pointer;
+
+  //      if(batch_count < 0)
+  //          return rocblas_status_invalid_size;
+
+        if(n <= 0 || incx <= 0 || batch_count <= 0)
```

Comment:
Batch count < 0 invalid_size is what I think is best, only quick success on == 0

Reply:
This is a L1 function. There are no incorrect argument checks in L1, only quick return.